pages/api/roadmap.ts

```diff
@@ -1,102 +1,102 @@
 
-const resolveChildren = (children: any[]): Promise<IssueData[]> => {
-  const resultArray: any = [];
-  let count = 0;
-
-  return new Promise((resolve, reject) => {
-    if (!_.isArray(children)) reject('Children is not an array. Is this a root issue?');
-    if (_.isArray(children) && children.length === 0) reject('Children array is empty. Is this a root issue?');
-
-    children.forEach((current) => {
-      getIssue(paramsFromUrl(current.html_url)).then((issueData) => {
-        checkForLabel(issueData);
-        resultArray.push({
-          ...issueData,
-          labels: issueData?.labels ?? [],
-          group: current.group,
-        });
-        count += 1;
-        if (count === children.length) {
-          resolve(resultArray);
-        }
-      });
-    });
-  });
+async function resolveChildren (children: ParserGetChildrenResponse[]): Promise<GithubIssueDataWithGroup[]> {
+  if (!Array.isArray(children)) {
+    throw new Error('Children is not an array. Is this a root issue?');
+  }
+
+  if (children.length === 0) {
+    throw new Error('Children array is empty. Is this a root issue?');
+  }
+
+  return Promise.all(children.map(async (child): Promise<GithubIssueDataWithGroup> => {
+    const urlParams = paramsFromUrl(child.html_url);
+    if (!urlParams) {
+      throw new Error('Could not parse URL');
+    }
+    const issueData = await getIssue(urlParams);
+    checkForLabel(issueData);
+    return {
+      ...issueData,
+      labels: issueData?.labels ?? [],
+      group: child.group
+    };
+  }));
 };
 
-const resolveChildrenWithDepth = async (children: ParserGetChildrenResponse[]) => {
-  return resolveChildren(children).then((issues) => {
-    const resultArray: any = [];
-    let count = 0;
-
-    return new Promise<IssueData>((resolve) => {
-      issues.forEach(async (issueData) => {
-        const childrenParsed = getChildren(issueData.body_html);
-        resultArray.push({
-          ...issueData,
-          labels: issueData.labels ?? [],
-          children: (childrenParsed.length > 0 && (await resolveChildren(childrenParsed))) || childrenParsed,
-        });
-        count += 1;
-        if (count === children.length) {
-          resolve(resultArray);
-        }
-      });
-    }).then((data) => {
-      return data;
-    }).catch((e) => {
-      return [];
-    })
-  });
+async function resolveChildrenWithDepth(children: ParserGetChildrenResponse[]): Promise<GithubIssueDataWithGroupAndChildren[]> {
+  try {
+    const issues = await resolveChildren(children);
+    return await Promise.all(issues.map(async (issueData): Promise<GithubIssueDataWithGroupAndChildren> => {
+      const childrenParsed = getChildren(issueData.body_html);
+      return {
+        ...issueData,
+        labels: issueData.labels ?? [],
+        children: (childrenParsed.length > 0 && (await resolveChildren(childrenParsed))) || childrenParsed,
+      }
+    }));
+  } catch (err) {
+    console.error('error:', err);
+    return [];
+  }
```

Comment:
I think if we start chopping these functions up into separate files and testing them we can improve stability and confidence drastically.